zulipterminal/ui_tools/buttons.py

```diff
@@ -131,7 +131,7 @@ def __init__(self, *, controller: Any, count: int) -> None:
 
 class PMButton(TopButton):
     def __init__(self, *, controller: Any, count: int) -> None:
-        button_text = f"Private messages [{primary_key_for_command('ALL_PM')}]"
+        button_text = f"Direct messages [{primary_key_for_command('ALL_PM')}]"
```

Comment:
Creating a set of tests for the button text for the four buttons in the top-left seems like a good idea. One test could be a simple button text check with expected value of the button text being hardcoded into the test. We can also create a config file (similiar to `config/keys` and `config/symbols`) to define the button text in a single place to avoid hardcoding the values in the text. Does this sound good?
> It'd be good to separate this out into a separate commit to make it clear it's unrelated to the code+test change of the other file, and also for this reason.

Ok, I'll add the test and create separate commits for the two changes plus their respective tests.

Reply:
I should have noted that it's worth checking if there's an open PR to add these tests - I don't remember one, but there are quite a few open PRs that need review or updating.

The expected button text could be one test for now.

The specific check I was thinking of was ensuring the layout of the text+hotkey has the correct format for all these buttons, since when making the change here there is an offset in the hotkey indicator. Can you see what I mean?

Regarding extracting the button text, I suspect keeping the text local to the UI element would be good in the short term. In the longer term this could certainly be refactored, but I would expect such a config to also handle the different behavior of the buttons; we'd end up with fewer Button classes for this, which would be generated at run-time from the details in the config. I'm not sure how feasible this is right now, due to different Button behaviors, though it would make it simpler to extend to other custom behavior if that was a useful feature.

Reply:
> I should have noted that it's worth checking if there's an open PR to add these tests - I don't remember one, but there are quite a few open PRs that need review or updating.

Ok, I'll have a look.

> The specific check I was thinking of was ensuring the layout of the text+hotkey has the correct format for all these buttons, since when making the change here there is an offset in the hotkey indicator. Can you see what I mean?
Thanks for pointing out the offset problem! We could have a regex check as a test too, then.

> Regarding extracting the button text, I suspect keeping the text local to the UI element would be good in the short term. In the longer term this could certainly be refactored, but I would expect such a config to also handle the different behavior of the buttons; we'd end up with fewer Button classes for this, which would be generated at run-time from the details in the config. I'm not sure how feasible this is right now, due to different Button behaviors, though it would make it simpler to extend to other custom behavior if that was a useful feature.

Yeah, this makes sense. This could definitely be done as a refactor to the button classes in the future, if adding custom buttons becomes more common. For now, I'll leave it as is.